src/main/java/dev/task/dndquest/service/impl/StoryLineServiceImpl.java

```diff
@@ -29,38 +32,57 @@ public class StoryLineServiceImpl implements StoryLineService {
     private final ChatClient aiClient;
     private final AiChatResponseParser parser;
     private final StoryLineMapper storyLineMapper;
-    private List<StoryLineShortResponseDto> availableStoryLines;
+    private final StoryLineRepository repository;
+    private List<StoryLineFullResponseDto> availableStoryLines;
 
     @Override
     @Cacheable(value = "ps", key = "#auth")
     public List<StoryLineShortResponseDto> getAvailableStoryLines(Authentication auth) {
         availableStoryLines = initAvailableStoryLines(FIRST_SERIAL_NUM);
-        return availableStoryLines;
+        return availableStoryLines.stream()
+                .map(storyLineMapper::mapFullToShortDto)
+                .toList();
     }
 
     @Override
     @CachePut(value = "ps", key = "#auth")
     public List<StoryLineShortResponseDto> addNewStoryLinesForAvailable(Authentication auth) {
         availableStoryLines.addAll(
                 initAvailableStoryLines(Integer.toString(availableStoryLines.size() + 1)));
-        return availableStoryLines;
+        return availableStoryLines.stream()
+                .map(storyLineMapper::mapFullToShortDto)
+                .toList();
     }
 
     @Override
     @Cacheable(value = "sl", key = "{#serialNumber, #auth}")
-    public StoryLineFullResponseDto selectStoryline(Integer serialNumber, Authentication auth) {
-        StoryLineShortResponseDto storyLine;
+    public StoryLineFullResponseDto selectStoryLine(Integer serialNumber, Authentication auth) {
+        StoryLineFullResponseDto storyLineDto;
         try {
-            storyLine = availableStoryLines.get(serialNumber - 1);
+            storyLineDto = availableStoryLines.get(serialNumber - 1);
         } catch (IndexOutOfBoundsException e) {
             throw new StoryLineNotFoundException();
         }
-        StoryLineFullResponseDto fullStoryLineDto = storyLineMapper.mapShortToFullDto(storyLine);
-        fullStoryLineDto.setStories(initStories(createFullStoryPrompt(storyLine)));
-        return fullStoryLineDto;
+        storyLineDto.setStories(initStories(createFullStoryPrompt(storyLineDto)));
+        return storyLineDto;
     }
 
-    private List<StoryLineShortResponseDto> initAvailableStoryLines(String storyLineSerialNumber) {
+    @Override
+    @Caching(evict = {
+            @CacheEvict(value = "sl", key = "{#serialNumber, #auth}"),
+            @CacheEvict(value = "ps", key = "#auth")})
+    public String startStoryLine(Integer serialNumber, Authentication auth) {
+        StoryLineFullResponseDto storyLineDto;
+        try {
+            storyLineDto = availableStoryLines.get(serialNumber - 1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new StoryLineNotFoundException();
+        }
+        repository.save(storyLineMapper.mapToEntity(storyLineDto));
+        return "done";
```

Comment:
This response brings no value. Is it is a subject for change?

Reply:
Yes, this endpoint is not complete, it will bind the storyline with PlayCharacter and Adventure and then redirect to the start of the story, but this functionality is not implemented yet.